pkg/agent/multicast/mcast_socket_linux.go

```diff
@@ -60,6 +60,18 @@ func (s *Socket) AddMrouteEntry(src net.IP, group net.IP, iif uint16, oifVIFs []
 	return multicastsyscall.SetsockoptMfcctl(s.GetFD(), syscall.IPPROTO_IP, multicastsyscall.MRT_ADD_MFC, mc)
 }
 
+func (s *Socket) GetMroutePacketCount(src net.IP, group net.IP) (pktCount uint32, err error) {
+	siocSgReq := multicastsyscall.SiocSgReq{
+		Src: [4]byte{src[0], src[1], src[2], src[3]},
```

Comment:
Do you think we can check IPv4 or IPv6 here, and parse the correct bytes from net.IP? In this way, it is easier to be compatible with IPv6, and we don't need to transform to IPv4 in the caller.